src/internal/middleware.js

```diff
@@ -40,8 +40,14 @@ export default function sagaMiddlewareFactory(options = {}) {
     throw new Error('`options.logger` passed to the Saga middleware is not a function!')
   }
 
-  if(options.onerror && !is.func(options.onerror)) {
-    throw new Error('`options.onerror` passed to the Saga middleware is not a function!')
+  if(options.onerror) {
+    if(isDev) log('warn', '`options.onerror` is deprecated. Use `options.onError` instead.')
+    options.onError = options.onerror
+    delete options['onerror']
```

Comment:
is there any particular reason of [bracket] notation here? wouldnt `delete options.onerror` be more natural?